### PR TITLE
Enumerable branch

### PR DESCRIPTION
**Pull request by mohammadumar28**
Project Completed 🎉 

Implemented all the methods required in the specifications.

### CODE REVIEW

**Comment by mohammadumar28** [comment]
I don't know how Stickler-ci came into Github actions

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
```
Comment (Ispirett) [inline enumerable.rb:10]: block = proc { |num| num < (0 + 9 ) / 2 } }
Testing Range(5, 50).each(&block)  is not equal to Range(5,50).my_each(&block)

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    return enum_for unless block_given?
+    i = 0
+    while i < self.length
+      yield(self[i])
+      i += 1     
+    end
+    self
+  end
+  
+  def my_each_with_index
```
Comment (Ispirett) [inline enumerable.rb:26]: block = proc { |num| num < (0 + 9 ) / 2 } }
Testing Range(5, 50).each_with_index(&block) is not equal to Range(5,50).my_each_with_index(&block)

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    return enum_for unless block_given?
+    i =0
+    while i < self.length
+      yield(self[i], i)
+      i += 1   
+    end
+    self
+  end
+
+  def my_select
```
Comment (Ispirett) [inline enumerable.rb:41]: block = proc { |num| num < (0 + 9 ) / 2 } }
Testing Range(5, 50).select(&block) is not equal to Range(5,50).my_select(&block)

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        i += 1
+      end
+      return value
+    end
+    
+  end
+
+# [].my_all? { |i| i < 4 } # False
+
+  def my_any?
```
Comment (Ispirett) [inline enumerable.rb]: false_block = proc { |num| num > 9 } }
Testing Range(5, 50).my_any?(&false_block) is not equal to Range(5,50).any?(&false_block)

File: enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    arr= []
+    x = 0
+    while x < self.length
+      arr << self[x] if yield(self[x]) == true
+      x += 1
+    end
+    arr
+  end
+
+  def my_all?
```
Comment (Ispirett) [inline enumerable.rb]: false_block = proc { |num| num > 9 } }
Testing Range(5, 50).my_all?(&false_block) is not equal to Range(5,50).all?(&false_block)

File: enumerable.rb [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        return true
+      end
+      i += 1
+    end
+    return value
+  end
+
+  # print [1, 3, 7, 5].my_any { |ele| ele%2 == 0 }
+
+  def my_none?
```
Comment (Ispirett) [inline enumerable.rb]: false_block = proc { |num| num > 9 } }
Testing Range(5, 50).my_none?(&false_block) is not equal to Range(5,50).none?(&false_block)

File: enumerable.rb [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    while i < self.length
+      if yield(self[i]) == true
+        return false
+      end
+      i += 1
+    end
+    return value
+  end
+
+  def my_count(arg = nil)
```
Comment (Ispirett) [inline enumerable.rb:195]: array.my_map { |num| num < 10 }) is not equal to (array.map { |num| num < 10 }

File: enumerable.rb [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        if self[i] == arg
+          counter += 1
+        end
+        i += 1
+      end
+      return counter
+    end
+  end
+
+  def my_inject (acc=nil)
```
Comment (Ispirett) [inline enumerable.rb]: search =  proc { |memo, word| memo.length > word.length ? memo : word } 
words.my_inject(&search))  is not equal to words.inject(&search)

**Pull request merged**